packages/babel-plugin-polyfill-regenerator/test/options.js

```diff
@@ -0,0 +1,20 @@
+import * as babel from "@babel/core";
+import polyfillRegenerator from "../lib";
+
+function transform(code, opts) {
+  return babel.transformSync(code, {
+    configFile: false,
+    plugins: [[polyfillRegenerator, opts]],
+  });
+}
+
+describe("targets", () => {
+  it("is not supported", () => {
+    expect(() =>
+      transform("code", {
+        method: "usage-pure",
+        targets: { chrome: 50 },
+      }),
+    ).toThrow(/targets/);
+  });
+});
```

Comment:
Can you also add a test that makes sure that top-level targets are supported? i.e.
```json5
{
  targets: { chrome: 50 },
  plugins: [polyfillRefenerator, { method: "usage-pure" }]
}
```
should not throw (because the top-level targets may be useful to other plugins).

Reply:
Good idea to double-check that!

Reply:
Hmm... that was a good catch, it does seem to (incorrectly) throw when top-level targets are used.

I'm actually not sure myself how to differentiate between top-level and plugin option within the arguments here...

Is there a way to differentiate simply that you know of? Or would that require some non-trivial refactoring to even implement?
I have a feeling it might be the latter 😅

Reply:
Added the failing tests here in a new commit for reference

Reply:
Added the fix as described [below](https://github.com/babel/babel-polyfills/pull/119#issuecomment-1076893006)